src/domains/User/UserProfile/UserProfile.tsx

```diff
@@ -19,37 +31,185 @@ const useStyles = makeStyles((theme) => ({
+    return [user, setUser] as const;
+}
+
 export default function UserProfile({ img }: Props) {
     const classes = useStyles();
+    const [user, setUser] = RequiredUser();
+    const initState: RegisterForm = { firstName: user.name.first, lastName: user.name.last, email: user.email.address, password: '', confirmPassword: '' };
+    const [snack] = useSnack();
+    const [regForm, errors, handleSubmit, handleChange] = useForm(initState);
+    const changeBothNames = React.useCallback(() => API.changeName(regForm.firstName, regForm.lastName), [regForm.firstName, regForm.lastName]);
+    const [sendName, isNameloading] = useEndpoint(changeBothNames, {
+        onSuccess: ({ data }) => {
+            setUser(data.user);
+            snack(`Updated ${data.user.name.first} ${data.user.name.last}!`);
+        },
+    });
+    const changeEmail = React.useCallback(() => API.changeEmail(regForm.email), [regForm.email]);
+    const [sendEmail, isEMload] = useEndpoint(changeEmail, {
+        onSuccess: ({ data }) => {
+            setUser(data.user);
+            snack(`Updated ${data.user.email.address}!`);
+        },
+    });
 
     return (
         <div className={classes.root}>
             <Grid container alignContent='center' spacing={2}>
-                <Grid component='span' item xs={12}>
-                    <Avatar src={img} alt='Profile Avatar' />
-                    {/* ROUTING: to page to upload new photo */}
-                </Grid>
-                <Grid component='span' item xs={12}>
-                    <TextField
-                        inputProps={{ 'aria-label': 'E-mail' }}
-                        label='E-mail'
-                        aria-label='E-mail'
-                        required
-                        type='text'
-                        value='TODO'
-                        onChange={() => {}}
-                    />
+                <Grid container spacing={2} id='userInfo'>
+                    <Grid component='span' item xs={12}>
+                        {/* ROUTING: to page to upload new photo */}
+                        <Avatar src={img} alt='Profile Avatar' />
+                    </Grid>
+                    
+                    <Grid component='span' item xs={12}>
+                        {/* <Form onSubmit={handleNameSubmit}> */}
+                        <Form onSubmit={handleSubmit(sendName)}>
+                            <FormContent>
+                                <TextField
+                                    inputProps={{ 'aria-label': 'First Name' }}
+                                    label='First Name'
+                                    aria-label='First Name'
+                                    id='fName'
+                                    required
+                                    type='text'
+                                    placeholder='Your First Name Here'
```

Comment:
Lets leave off the placeholder here and other places since since we have a good label.

I don't really have a concrete reason to give you for removing the placeholder other than stylistic consistency (we don't use them anywhere else really).  Plus, I'm not sure it's part of the material ui spec even though it's available.  More here: https://material.io/components/text-fields (if you ctrl + f placeholder it's nowhere to be found and I didn't notice it in any examples when I quickly scanned through)

Reply:
Ah gotcha, removed it, I thought since it was an input field, that it would support it.
Referencing: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefplaceholder